spek-runtime/src/jvmMain/kotlin/org/spekframework/spek2/runtime/JvmDiscoveryContextFactory.kt

```diff
@@ -12,6 +12,10 @@ import kotlin.reflect.full.primaryConstructor
 import kotlin.streams.toList
 
 object JvmDiscoveryContextFactory {
+    private val scanParallelism by lazy {
+        Runtime.getRuntime().availableProcessors()
```

Comment:
Lazy shouldn't be necessary here?

If I use `gradle --parallel` with 8 cores this will start 8*8 threads to scan the classpath?

Reply:
I was about to link this, and yeah it probably will. I'm still thinking of a way to make this configurable.

Reply:
Maybe via a jvm property? https://docs.gradle.org/current/dsl/org.gradle.api.tasks.testing.Test.html#org.gradle.api.tasks.testing.Test:jvmArgs WDYT?

Reply:
It sucks that you can't pass configs from gradle to JUnit Platform test engines.

Reply:
I think this should be sequential by default and be configurable like you did with the `timeout` property:
`systemProperty("SPEK_TIMEOUT", 0) // disable test timeout`.
Junit5 itself mentions this in https://junit.org/junit5/docs/current/user-guide/#running-tests-build-gradle-config-params

Reply:
Ahh that makes sense - I don't have time now, but I'll look into it in the weekend. Thanks!